aptos-move/aptos-vm-types/src/output.rs

```diff
@@ -17,23 +17,30 @@ use aptos_types::{
     transaction::{TransactionAuxiliaryData, TransactionOutput, TransactionStatus},
     write_set::WriteOp,
 };
+use derivative::Derivative;
 use move_core_types::{
     value::MoveTypeLayout,
     vm_status::{StatusCode, VMStatus},
 };
+use move_vm_runtime::execution_tracing::Trace;
 use move_vm_types::delayed_values::delayed_field_id::DelayedFieldID;
-use std::collections::BTreeMap;
+use std::{collections::BTreeMap, mem};
 
 /// Output produced by the VM after executing a transaction.
 ///
 /// **WARNING**: This type should only be used inside the VM. For storage backends,
 /// use `TransactionOutput`.
-#[derive(Debug, Clone, Eq, PartialEq)]
+#[derive(Derivative)]
+#[derivative(Debug, Clone, Eq, PartialEq)]
 pub struct VMOutput {
     change_set: VMChangeSet,
     module_write_set: ModuleWriteSet,
     fee_statement: FeeStatement,
     status: TransactionStatus,
+    /// Trace of the user transaction payload execution in Move VM. Trace is always created as
+    /// empty, and users have to set it manually after execution.
+    #[derivative(PartialEq = "ignore", Debug = "ignore")]
```

Comment:
is ignoring for PartialEq appropriate? (depends on use cases for equality), where do we even compare outputs, could be just testing?

Reply:
Comparison is only for outputs, also if you run sequential vs parallel trace can differ, so it cannot be part of this top-level comparison